Sources/MultiSets/CountedSet/CountedSet.swift

```diff
@@ -0,0 +1,41 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+/// An unordered, counted multiset.
+@frozen
+public struct CountedSet<Element: Hashable>: RawRepresentable {
+  // Allows internal setter to be referenced from inlined code
+  @usableFromInline
+  internal var _storage = RawValue()
+
+  @inlinable @inline(__always)
+  public var rawValue: [Element: UInt] { _storage }
```

Comment:
I originally used a `Dictionary<Element, Int>` as a backing store, but decided to change it purely to eliminate confusion if a signed multiset was ever added. No one can look at a `UInt` and think it can hold a negative value.

Reply:
Any thoughts about whether that is sufficient justification for using `UInt`?

Reply:
I don't disagree with that, but I feel using UInt here would be diverging too much from established API design precedents. The dice have fallen this way long ago, and it's too late to change our minds about it.

(FWIW, I do like that in Swift we tend not to do arithmetic on UInt values, or deal with signed/unsigned conversions in regular code.)